src/MutableArithmetics.jl

```diff
@@ -1,5 +1,128 @@
+#  Copyright 2019, Gilles Peiffer, Benoît Legat, Sascha Timme, and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#############################################################################
+
 module MutableArithmetics
 
-greet() = print("Hello World!")
+using LinearAlgebra, Base.GMP.MPZ
+
+import Base.*, Base.promote_op, Base.==
+import LinearAlgebra.mul!
+
+export DummyBigInt
+
+
+"""
+    AbstractMutable
+
+Generic supertype for types
+implementing the MutableArithmetics interface.
+"""
+abstract type AbstractMutable end
+
+"""
+    DummyBigInt
+
+Mutable wrapper type around `BigInt`.
+"""
+struct DummyBigInt <: AbstractMutable
+    data::BigInt
+end
+
+# For convenience.
+Base.zero(::Type{DummyBigInt}) = DummyBigInt(0)
+Base.promote_op(matprod, ::Type{DummyBigInt}, ::Type{DummyBigInt}) = DummyBigInt
```

Comment:
You shouldn't have to implement this, `promote_op` will automatically work if you define `*` and `+` between `DummyBigInt`